mongo/integration/unified_runner_events_helper_test.go

```diff
@@ -24,57 +24,89 @@ import (
 // pool events.
 
 var (
-	eventTypesMap = map[string]string{
+	poolEventTypesMap = map[string]string{
 		"PoolClearedEvent": event.PoolCleared,
 	}
 	defaultCallbackTimeout = 10 * time.Second
+	sdamEventCount         = make(map[string]int)
```

Comment:
While `poolEventCount` can be stored on `unifiedRunnerEventMonitor`, `sdamEventCount` and its associated mutex cannot be. If `sdamEventCount` were stored on `unifiedRunnerEventMonitor`, `handleServerDescriptionChanged` would have to be a method on `unifiedRunnerEventMonitor` to access `sdamEventCount` and the mutex. However, on L51, `event.ServerMonitor` requires a _function_ for the value of `ServerDescriptionChanged` (the event handler for `ServerDescriptionChangedEvent`s). There'd be no way to refer to `handleServerDescriptionChanged` if it were a method. So, I simply reset the global `sdamEventCount` on each call to `newUnifiedRunnerEventMonitor`. Let me know if this doesn't make sense.

Reply:
Clever! Yes a closure would work; sometimes I forget how functional Go can be 😆 Thanks.